### PR TITLE
feat: add playwright e2e testing

### LINKED ISSUE #539: test: Add E2E testing infrastructure
### Type of feature

Select the type of feature request, the lowercase should also be the PR prefix.

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [x] 🔁 CI
- [] 📦 Chore (Release)
- [ ] ⏩ Revert

### Current behavior

Is your feature request related to a problem? Please describe.

### Suggested solution

Describe the solution you'd like.

This effort:
- Sets up Cypress dependencies (https://cypress.io)
- Adds initial setup to run E2E tests against the insights page
- Sets up smoke tests to ensure critical functionality of insights pages (dashboard, repos, contributors)
- Sets up additional tests less critical tests
- Sets up structure for future testing efforts
- Documents running E2E test setup and practices

### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](https://docs.opensauced.pizza/contributing/code-of-conduct/).

- [ ] I agree to follow this project's Code of Conduct

### Contributing Docs

If you plan on contributing code please read - [Contribution Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/)

- [ ]  I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
Installs & setups Playwright framework and introduces a simple starter e2e test.

## Related Tickets & Documents

fixes #539 
## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,13 @@
     "prebuild": "node generate-netlify-env.js",
     "build": "next build",
     "start": "next start",
-    "test": "jest",
+    "test": "npm run test:jest && npm run test:e2e",
```
Comment (brandonroberts) [inline package.json:34]: I think  `e2e` should be its on script and separate from the `npm run test` command. Is this not picked up automatically by the actions to run E2E tests?

```suggestion
    "test": "npm run test:jest && npm run test:e2e",
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Deadreyo) [reply package.json:34]: https://github.com/open-sauced/open-sauced/blob/main/.github/workflows/development.yml
only `npm run test` is executed in the CI.

Comment (brandonroberts) [reply package.json:34]: I see. I associate running `npm run test` with unit tests. Not a blocker for this PR though

**Review by brandonroberts — APPROVED** [review]

**Review by 0-vortex — APPROVED** [review]
LGTM! 👍

**Pull request merged**